app/src/main/java/fr/free/nrw/commons/depictions/GridViewAdapter.java

```diff
@@ -0,0 +1,119 @@
+package fr.free.nrw.commons.depictions;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.TextView;
+
+import androidx.annotation.Nullable;
+
+import com.facebook.drawee.view.SimpleDraweeView;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import fr.free.nrw.commons.Media;
+import fr.free.nrw.commons.R;
+
+/**
+ * Adapter for Items in DepictionDetailsActivity
+ */
+public class GridViewAdapter extends ArrayAdapter {
+
+        private List<Media> data;
+
+        public GridViewAdapter(Context context, int layoutResourceId, List<Media> data) {
+            super(context, layoutResourceId, data);
+            this.data = data;
+        }
+
+        /**
+         * Adds more item to the list
+         * Its triggered on scrolling down in the list
+         * @param images
+         */
+        public void addItems(List<Media> images) {
+            if (data == null) {
+                data = new ArrayList<>();
+            }
+            data.addAll(images);
+            notifyDataSetChanged();
+        }
+
+        /**
+         * Check the first item in the new list with old list and returns true if they are same
+         * Its triggered on successful response of the fetch images API.
+         * @param images
+         */
+        public boolean containsAll(List<Media> images){
+            if (images == null || images.isEmpty()) {
+                return false;
+            }
+            if (data == null) {
+                data = new ArrayList<>();
+                return false;
+            }
+            if (data.size() <= 0) {
+                return false;
+            }
+            String fileName = data.get(0).getFilename();
```

Comment:
Is this intentional

Reply:
How could I answer that?

Reply:
I guess the task here is to find out whether this line is correct or not.
At first look it is suspicious, so it is something that PR reviewers need to point out.

Reply:
```
 /**
         * Check the first item in the new list with old list and returns true if they are same
         * Its triggered on successful response of the fetch images API.
         * @param images
         */
```
It was 100% intended apparently